### PR TITLE
Improve the comment for 2FA filter in admin panel

### PR DESCRIPTION
**Pull request by Gusted**
- Better TODO message.

### CODE REVIEW

File: models/user/user.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ type User struct {
 	EmailNotificationsPreference string `xorm:"VARCHAR(20) NOT NULL DEFAULT 'enabled'"`
 	Passwd                       string `xorm:"NOT NULL"`
 	PasswdHashAlgo               string `xorm:"NOT NULL DEFAULT 'argon2'"`
+	Is2faEnabled                 bool   `xorm:"NOT NULL DEFAULT false column_name 'is_2fa_enabled'"`
```
Comment (lunny) [inline models/user/user.go]: When this field is updated?

**Comment by wxiaoguang** [comment]
That `TODO` comes from me (I added the filter)

If there is no performance problem, I think we can postpone the introducing of new fields.

It seems that `2FA` system should to be refactored also.

File: models/user/user.go (hunk 1 above)
Comment (Gusted) [reply models/user/user.go]: Updated.

**Comment by Gusted** [comment]
> That `TODO` comes from me (I added the filter)

Ah, IDE showed that last activity came from Lunny(so I just assumed)



> If there is no performance problem, I think we can postpone the introducing of new fields.
> 
> It seems that `2FA` system should to be refactored also.

Last time I checked using JOIN's aren't quite good for the performance lookups.

**Comment by wxiaoguang** [comment]
> Last time I checked using JOIN's aren't quite good for the performance lookups.

Yep, although its performance not good, as long as it doesn't harm now so it's not too bad 😂.

Since we are refactoring u2f/webauthn now, I have a feeling that 2FA mechanism should be improved after these refactoring.

**Comment by lunny** [comment]
More and more columns in user table is not a good idea imo.

**Comment by singuliere** [comment]
I agree that adding a new field is not worth the gain. The TODO comment should be updated to not suggest adding the field so that someone else does not try to implement what turns out to be something that is not desirable.

**Comment by lunny** [comment]
Since user setting table has been added, I think this PR could do some adjustment to use that table but not create a new column.

**Comment by Gusted** [comment]
I've for now changed the TODO

File: models/user/search.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ func (opts *SearchUserOptions) toSearchQueryBase() *xorm.Session {
 	}
 
 	// 2fa filter uses LEFT JOIN to check whether a user has a 2fa record
-	// TODO: bad performance here, maybe there will be a column "is_2fa_enabled" in the future
+	// TODO: Add a "is_2fa_enabled" column to the user' setting table.
```
Comment (wxiaoguang) [inline models/user/search.go]:
```suggestion
	// While using LEFT JOIN, sometimes the performance might not be good, but it won't be a problem now, such SQL is seldom executed.
	// There are some possible methods to refactor this SQL in future when we really need to optimize the performance (but not now):
	// (1) add a column in user table (2) add a setting value in user_setting table (3) use search engines (bleve/elasticsearch)
```

**Review by wxiaoguang — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**